lib/ansible/module_utils/facts/hardware/darwin.py

```diff
@@ -85,38 +85,43 @@ def get_cpu_facts(self):
 
     def get_memory_facts(self):
         memory_facts = {
-            'memtotal_mb': int(self.sysctl['hw.memsize']) // 1024 // 1024
+            'memtotal_mb': int(self.sysctl['hw.memsize']) // 1024 // 1024,
+            'memfree_mb': 0,
         }
 
         total_used = 0
         page_size = 4096
-        vm_stat_command = get_bin_path('vm_stat')
-        rc, out, err = self.module.run_command(vm_stat_command)
-        if rc == 0:
-            # Free = Total - (Wired + active + inactive)
-            # Get a generator of tuples from the command output so we can later
-            # turn it into a dictionary
-            memory_stats = (line.rstrip('.').split(':', 1) for line in out.splitlines())
-
-            # Strip extra left spaces from the value
-            memory_stats = dict((k, v.lstrip()) for k, v in memory_stats)
-
-            for k, v in memory_stats.items():
-                try:
-                    memory_stats[k] = int(v)
-                except ValueError as ve:
-                    # Most values convert cleanly to integer values but if the field does
-                    # not convert to an integer, just leave it alone.
-                    pass
-
-            if memory_stats.get('Pages wired down'):
-                total_used += memory_stats['Pages wired down'] * page_size
-            if memory_stats.get('Pages active'):
-                total_used += memory_stats['Pages active'] * page_size
-            if memory_stats.get('Pages inactive'):
-                total_used += memory_stats['Pages inactive'] * page_size
-
-        memory_facts['memfree_mb'] = memory_facts['memtotal_mb'] - (total_used // 1024 // 1024)
+        try:
+            vm_stat_command = get_bin_path('vm_stat')
+        except ValueError:
+            vm_stat_command = None
+        if vm_stat_command:
```

Comment:
I'm just going to leave this comment once but it applies to most of the ports:

I think this is fine for a port but the next time this is refactored, it should probably look more like this:
``` python
try:
    vm_stat_command = get_bin_path('vm_stat')
except ValueError:
    # The vm stat was not found.  Return early with the values we were able to determine
    return memory_facts

rc, out, err = self.module.run_command(vm_stat_command)
[...]
```

returning early is better code as it allows the reader of the code to stop worrying about what the other conditional path might do later on in the function.  (Learned this from Jack's talk at Pycon 2018)

Reply:
Agree. I like this much better. You can stop reading at the `return` rather than having to read through the rest of the function.